File: java/src/main/java/com/danifoldi/dml/type/DmlBoolean.java
package com.danifoldi.dml.type;

import com.danifoldi.dml.DmlSerializer;

public class DmlBoolean extends DmlCommentableValue {
    private boolean value;

    public DmlBoolean(boolean value) {
        this.value = value;
    }

    public boolean value() {
        return this.value;
    }

    public void value(boolean value) {
        this.value = value;
    }

    @Override
    public String serialize(int indent) {
        return DmlSerializer.serializeComment(comment(), indent) + DmlSerializer.indent(String.valueOf(value), indent);
    }

    @Override
    public String toString() {
        return "DmlBoolean{" + value + "}";
    }
}

File: javascript/src/types/DmlBoolean.js
import DmlCommentableValue from './DmlCommentableValue.js'
import DmlSerializer from '../DmlSerializer.js'

class DmlBoolean extends DmlCommentableValue {
  #value

  constructor(value) {
    super()
    this.#value = value
  }

  value(value) {
    if (typeof value === 'undefined') {
      return this.#value
    }

    this.#value = value
  }

  serialize(indent) {
    return DmlSerializer.serializeComment(this.comment(), indent)
      + DmlSerializer.indent(this.#value ? 'true' : 'false', indent)
  }
}

export default DmlBoolean
